LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file and sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its LocalFolder directory.
+* Support web apps not installed and managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+* Introduce a non-standard Web API
+
+## Proposed Solution 
+
+### Existing behavior
+
+`navigator.storage.getDirectory()` is an existing API that returns a promise to a FileSystemDirectoryHandle representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
+
+### New behavior
+
+Within a Microsoft Store PWA context, the directory handle for OPFS root will contain an additional entry that represents the app's LocalFolder location. This entry can be retrieved as a `FileSystemDirectoryHandle` by calling `.entries()` or `.getDirectory(...)` on the OPFS root directory handle.
+
+The LocalFolder directory handle and its contents will have properties different from other OPFS directory handles - they will only have read and delete permissions instead of read and write permissions.
+
+```JS
+    // .getDirectory()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+    }
+```
+
+```JS
+    // .entries()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        for await (const [key, value] of opfsRoot.entries()) {
+            if (key == `microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt`) {
+                // ...
+            }
+        }
+    }
+```
+
+### Deleting files
+
+LocalStorage contents can be cleared by calling `.removeEntry(...)` from the OPFS root directory handle.
+```JS
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        await opfsRoot.removeEntry("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", { recursive: true }); 
+    }
+```
+
+Alternatively, contents in LocalFolder can be deleted by calling `remove()` on file and directory handles directory.
+
+```JS
+    let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+
+    for await (const [key, value] of localStorage.entries()) {
+        value.remove();
+    }
+```
+
+### Entry name
+
+The LocalFolder entry within the OPFS root directory can be found under the name `microsoft_store_app_local_folder{PFN}` where `{PFN}` is the Package Family Name of the app.
```

Comment:
This naming is inconsistent with the examples above (missing an underscore before the `{PFN}`:

```suggestion
The LocalFolder entry within the OPFS root directory can be found under the name `microsoft_store_app_local_folder_{PFN}` where `{PFN}` is the Package Family Name of the app.
```